quantinuum-hugr/src/types.rs

```diff
@@ -40,8 +52,8 @@ pub enum EdgeKind {
     ControlFlow,
     /// Data edges of a DDG region, also known as "wires".
     Value(Type),
-    /// A reference to a static value definition.
-    Static(Type),
+    /// A reference to a static value definition - Const or Function
+    Static(StaticEdgeData),
```

Comment:
This is the bit I'm least sure about. The other possibility is to have two distinct EdgeKinds (i.e. one new) - ideally I'd call them Const (rather than Static) and Function, or perhaps StaticValue and StaticFunction. We even had ConstE edges in the spec at one point....